src/main/scala/bitboard/OpaqueBitboard.scala

```diff
@@ -52,6 +52,20 @@ trait OpaqueBitboard[A](using A =:= Long) extends TotalWrapper[A, Long]:
     // remove the first non empty position
     def removeFirst: A = (a.value & (a.value - 1L)).bb
 
+    // check if the intersection of all bitboards is not empty
+    def sharedAny(o: Long, xs: Long*): Boolean =
+      (a & xs.foldLeft(o)(_ & _)).nonEmpty
+
+    // check if the intersection of all bitboards is not empty
+    def sharedAny[B](o: B, xs: B*)(using sr: BitboardRuntime[B]): Boolean =
+      (a & xs.foldLeft(sr(o))(_ & sr(_))).nonEmpty
+
+    // check if the intersection of all bitboards is empty
+    def sharedNone(o: Long, xs: Long*): Boolean = !sharedAny(o, xs*)
```

Comment:
Is this function needed?  Couldn't you just use `sharedAny` (or a different name)?

Reply:
I think if we go with `sharedAny`, `sharedNone` is a nice companion function. Just like `isEmpty` and `nonEmpty`.